js/katex-a11y.js

```diff
@@ -0,0 +1,485 @@
+};
+
+var openMap = {
+    "|": "open vertical bar",
+    ".": ""
+};
+
+var closeMap = {
+    "|": "close vertical bar",
+    ".": ""
+};
+
+var binMap = {
+    "+": "plus",
+    "-": "minus",
+    "\\pm": "plus minus",
+    "\\cdot": "dot",
+    "*": "times",
+    "/": "divided by",
+    "\\times": "times",
+    "\\div": "divided by",
+    "\\circ": "circle",
+    "\\bullet": "bullet"
+};
+
+var relMap = {
+    "=": "equals",
+    "\\approx": "approximately equals",
+    "\\neq": "does not equal",
+    "\\ne": "does not equal",
+    "\\geq": "is greater than or equal to",
+    "\\ge": "is greater than or equal to",
+    "\\leq": "is less than or equal to",
+    "\\le": "is less than or equal to",
+    ">": "is greater than",
+    "<": "is less than",
+    "\\leftarrow": "left arrow",
+    "\\Leftarrow": "left arrow",
+    "\\rightarrow": "right arrow",
+    "\\Rightarrow": "right arrow",
+    ":": "colon"
+};
+
+var buildString = function(str, type, a11yStrings) {
+    if (!str) {
+        return;
+    }
+
+    var ret;
+
+    if (type === "open") {
+        ret = (str in openMap ? openMap[str] : stringMap[str] || str);
+    } else if (type === "close") {
+        ret = (str in closeMap ? closeMap[str] : stringMap[str] || str);
+    } else if (type === "math") {
+        ret = mathMap[str] || str;
+    } else if (type === "bin") {
+        ret = binMap[str] || str;
+    } else if (type === "rel") {
+        ret = relMap[str] || str;
+    } else {
+        ret = stringMap[str] || str;
+    }
+
+    // If nothing was found and it's not a plain string or number
+    if (ret === str && !/^\w+$/.test(str)) {
+        // This is likely a case that we'll need to handle
+        throw "KaTeX a11y " + type + " string not found: " + str;
+    }
+
+    // If the text to add is a number and there is already a string
+    // in the list and the last string is a number then we should
+    // combine them into a single number
+    if (/^\d+$/.test(ret) && a11yStrings.length > 0 &&
+            /^\d+$/.test(a11yStrings[a11yStrings.length - 1])) {
+        a11yStrings[a11yStrings.length - 1] += ret;
+
+    } else if (ret) {
+        a11yStrings.push(ret);
+    }
+};
+
+var buildRegion = function(a11yStrings, callback) {
+    var region = [];
+    a11yStrings.push(region);
+    callback(region);
+};
+
+var typeHandlers = {
+    accent: function(tree, a11yStrings) {
+        buildRegion(a11yStrings, function(a11yStrings) {
+            buildA11yStrings(tree.value.base, a11yStrings);
+            a11yStrings.push("with");
+            buildA11yStrings(tree.value.accent, a11yStrings);
+            a11yStrings.push("on top");
+        });
+    },
+
+    bin: function(tree, a11yStrings) {
+        buildString(tree.value, "bin", a11yStrings);
+    },
+
+    close: function(tree, a11yStrings) {
+        buildString(tree.value, "close", a11yStrings);
+    },
+
+    color: function(tree, a11yStrings) {
+        var color = tree.value.color.replace(/katex-/, "");
```

Comment:
@xymostech I'm not sure if there really is a way to ideally handle custom colors. I have a test case for this and right now:

```
\blue{a}\color{#0f0}{b}\color{red}{c}
```

Is turned in to:

```
start color blue, a, end color blue, start color #0f0, b, end color #0f0, start color red, c, end color red
```

Which is probably as good as we're going to get. I mean communicating a name is probably better than a weird hex thing, but at least the hexes will be consistent? It's mostly being used as a way to denote that some portion of the formula is different from another portion - and I think it will succeed in that regard.